File: IMT2021545_CPP_Lab5/Mattress.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;
#include"Mattress.h"

void Mattress::set_valmat(string ty,int wi,int le,int delt,int pr)
{
  // setting all the relevant data
  type=ty;
  width=wi;
  length=le;
  deltime=delt;
  price=pr;
}

// getters
int Mattress::getDeltime()
{
  return deltime;
}

int Mattress::getWidth()
{
  return width;
}

int Mattress::getLength()
{
  return length;
}

// method to print all the info
void Mattress::getInfo() {
    string inf = "Mattress " + getType() + " " + to_string(getWidth()) + " " + to_string(getLength()) + " " + to_string(getDeltime()) + " " + to_string(getPrice());
    cout << inf << endl;
}

File: IMT2021545_Lab5/Mattress.java
public class Mattress extends Product {
    
    private int length;
    private int width;
    private int deltime;
    //setter method
    void setVals(String type,int w,int l,int d,int p)
    {
        length=l;
        width=w;
        deltime=d;
        super.setVals(type,p);
    }
    //getter methods
    int getLength(){
        return length;
    }

    int getWidth(){
        return width;
    }
    
    int getDeltime(){
        return deltime;
    }
}